scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -907,6 +907,7 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
   def acceptStatSep(): Unit = token match {
     case LF() | LFLF() => next()
     case _ if in.observeOutdented() =>
+    case t if t.is[EndMarkerIntro] =>
```

Comment:
Is this needed? Outdent should be always before end marker and from what I see we don't need it below.

Reply:
For empty template body, there is no indent/outdent
```scala
object Foo: // <- here at colon  we call `observeIndented` while the next line doesn't have any indentation
end Foo
```

Reply:
Generally, it should end on the LF after `:`, but this doesn't seem to happen and we are skipping the LF. That would be the underlying issue here. However, not sure how to fix it properly, so it's ok for now.

Reply:
and would this work:
```
object Foo:
// comment
end Foo
```

Reply:
@kitbellew yep. Parser ignores comments in the same way as whitespaces